src/lib/state/web3.ts

```diff
@@ -12,6 +13,14 @@ export type Web3ReactState = [Connector, Web3ReactHooks]
 
 export const urlAtom = atomWithDefault<Web3ReactState>(() => EMPTY_CONNECTOR)
 export const injectedAtom = atomWithDefault<Web3ReactState>(() => EMPTY_CONNECTOR)
+export const providerAtom = atom<Web3ReactState>((get) => {
+  const url = get(urlAtom)
+  const injected = get(injectedAtom)
+  if (injected[0] !== EMPTY) {
+    return get(injectedAtom)
```

Comment:
`return injected`.

nit: this could also be condensed using a ternary in the return statement